app/services/action_builder.rb

```diff
@@ -44,41 +44,22 @@ def member
     return @user if @user.present?
     @user = existing_member || Member.new(email: @params[:email])
 
-    update_member_fields
```

Comment:
I'm sorry if this refactor makes the changes a little harder to grasp. But I do think it was the best way to handle this: I needed to reuse some of the logic in this class, and I think this was the simpler way I could do it. I just extracted the logic to update members to separate service doing minimal changes to the ActionBuilder class.